### DISCUSSION TITLE
想把代码加进论文附录里面，但是遇到两个问题

### DISCUSSION THREAD

**Discussion by jpDeng111** [sjtug/SJTUThesis#668] [提问] [answered]
操作系统：Win10
模板：SJTUThesis develop
套用模板中的这段代码，发现（1）页码的“共56页”变成了“共XX页”
并且代码并没有完全插进去

```tex
\begin{codeblock}[language=Python]
#include <stdio.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>

int main() {
  pid_t pid;

  switch ((pid = fork())) {
  case -1:
    printf("fork failed\n");
    break;
  case 0:
    /* child calls exec */
    execl("/bin/ls", "ls", "-l", (char*)0);
    printf("execl failed\n");
    break;
  default:
    /* parent uses wait to suspend execution until child finishes */
    wait((int*)0);
    printf("is completed\n");
    break;
  }

  return 0;
}
\end{codeblock}
```

**Comment by skyzh** [comment, marked as answer]
您提供的代码可以正常插入。建议检查其他地方是否有问题。

<img width="564" alt="image" src="https://user-images.githubusercontent.com/4198311/120097254-19477000-c162-11eb-977d-67197fb55276.png">

**Reply by jpDeng111** [reply]
嗯嗯对的，就是如果太长可能放不进去了，比如这段（一开始没放是怕放上来会影响查重率，不过已经查重完了，管它呢）
万一解决不了也没事，只是因为听说答辩老师会看论文厚度，周围同学都附上代码了，所以也想附上代码来着

**Comment by jpDeng111** [comment]
嗯嗯对的，就是如果太长可能放不进去了，比如这段（一开始没放是怕放上来会影响查重率，不过已经查重完了，管它呢）
import os
from matplotlib import pyplot
import cv2
from functools import reduce
import numpy as np
import matplotlib.pyplot as plt
import scipy.interpolate as spi
from scipy import signal #滤波等
from skimage import measure,data_dir,io,color
import random
from scipy.interpolate import CubicSpline
import math
import lvreader as lv
import seaborn as sns
#判断文件是否是图片，如果不是，就返回0.如果图片形式都统一为一种，那么就可以直接把它写进主程序里面

def is_img(ext):
 ext = ext.lower()
 if ext == '.vc7':
     print('yes')#测试有没有运行检查图片功能这个函数
     return True
 elif ext == '.im7':
     print('hi')#测试有没有执行
     return True
 elif ext == '.png':
     print('hi1')#测试有没有执行
     return True
 elif ext == '.jpg':
     print('hi2')#测试有没有执行
     return True
 elif ext == '.jpeg':
     return True
 elif ext == '.bmp':
     return True
 else:
     return False

#函数定义：输入contours与计算某一个曲率点数，选取contours中最大的边界，根据点数计算曲率，输出C0为最长边界的曲率值
def curvature(contour,dianshu):
        hang0=contour.shape[0]#获得行数
        ax.plot(contour[:, 1], contour[:, 0], linewidth=2)
        C0=np.zeros((hang0-dianshu+1)*1)#用于储存曲率数值
        m=int(dianshu/2-0.5)#掐头去尾的点数
        for p in range(m,hang0-m):
            x=contour[p-m:p+m+1,0]
            y=contour[p-m:p+m+1,1]
            t=np.zeros_like(x,dtype=float)
            T=np.zeros_like(x,dtype=float)
            for i in range(dianshu-1):
                b=((y[i+1]-y[i])**2+(x[i+1]-x[i])**2)**0.5
                t[i+1]=b
                T[i+1]=T[i]+t[i+1]#参数T随xy变化
            #print(T)
            csx = CubicSpline(T, x)#计算x随参数T的三次样条插值
            csy = CubicSpline(T, y)
            #print(csx(T, 1))#x插值函数在数据点的一阶导数
            #print(csx(T, 2))
            #print(csy(T, 1))
            #print(csy(T, 2))
            dex=np.zeros(2)#排序为一阶二阶导数
            dey=np.zeros(2)
            for i in range(2):
                mx=csx(T,i+1)
                n=m
                dex[i]=mx[int(n)]
                my=csy(T,i+1)
                dey[i]=my[int(n)]
            c0=(dex[0]*dey[1]-dex[1]*dey[0])/(dex[0]**2+dey[0]**2)**1.5#曲率计算
            C0[p-m]=c0#存储一条边缘的曲率到C0中
        return C0
def myFunc(e):
    #长度排序函数
      return len(e)



#导入图片路径，注意像素值为整数
filepath=r"E:/graduatepap/codes/testfile6/PIVexm"#图片路径
#统计曲率的张数,我觉得可以读出来目标文件夹里面有多少张照片
line_number=2#用于计算曲率的边界数
dianshu=11#计算曲率所用点数

**Reply by skyzh** [reply]
和长度没有关系，这边可以处理跨页的代码。

<img width="861" alt="image" src="https://user-images.githubusercontent.com/4198311/120097920-97594600-c165-11eb-9d95-726a276e38d3.png">

你提供的代码似乎会导致 LaTeX 出错。可以考虑去掉注释的部分。

```
./contents/app_flow_chart.tex:76: You can't use `macro parameter character #' i
n horizontal mode.
```
